hazelcast/proxy/map.py

```diff
@@ -48,80 +97,194 @@ def add_entry_listener(self, include_value=False, key=None, predicate=None, adde
         def handle_event_entry(**_kwargs):
             event = EntryEvent(self._to_object, **_kwargs)
             if event.event_type == EntryEventType.added:
-                added(event)
+                added_func(event)
             elif event.event_type == EntryEventType.removed:
-                removed(event)
+                removed_func(event)
             elif event.event_type == EntryEventType.updated:
-                updated(event)
+                updated_func(event)
             elif event.event_type == EntryEventType.evicted:
-                evicted(event)
+                evicted_func(event)
             elif event.event_type == EntryEventType.evict_all:
-                evict_all(event)
+                evict_all_func(event)
             elif event.event_type == EntryEventType.clear_all:
-                clear_all(event)
+                clear_all_func(event)
             elif event.event_type == EntryEventType.merged:
-                merged(event)
+                merged_func(event)
             elif event.event_type == EntryEventType.expired:
-                expired(event)
+                expired_func(event)
 
         return self._start_listening(request, lambda m: map_add_entry_listener_codec.handle(m, handle_event_entry),
                                      lambda r: map_add_entry_listener_codec.decode_response(r)['response'])
 
     def add_index(self, attribute, ordered=False):
+        """
+        Adds an index to this map for the specified entries so that queries can run faster.
+
+        Example:
+            Let's say your map values are Employee objects.
+                >>> class Employee(IdentifiedDataSerializable):
+                >>>     active = false
+                >>>     age = None
+                >>>     name = None
+                >>>     #other fields
+                >>>
+                >>>     #methods
+
+            If you are querying your values mostly based on age and active then you should consider indexing these
```

Comment:
If you query your values mostly based on age and active fields, you should consider indexing these.